k8s/src/test/scala/io/buoyant/k8s/IngressCacheTest.scala

```diff
@@ -1,12 +1,228 @@
+        "uid": "078c9288-813d-11e7-b89b-080027a996b8",
+        "resourceVersion": "58096",
+        "generation": 1,
+        "creationTimestamp": "2017-08-14T22:07:57Z",
+        "annotations": {
+          "ingress.kubernetes.io/rewrite-target": "/"
+        }
+      },
+      "spec": {
+        "rules": [
+          {
+            "http": {
+              "paths": [
+                {
+                  "path": "/nginx-only",
+                  "backend": {
+                    "serviceName": "nginx",
+                    "servicePort": 8080
+                  }
+                },
+                {
+                  "path": "/shared",
+                  "backend": {
+                    "serviceName": "nginx",
+                    "servicePort": 8081
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      },
+      "status": {
+        "loadBalancer": {}
+      }
+    }
+  ]
+}"""
+
+  val ingressResourceListWithOneIngress = """
+{
+  "kind": "IngressList",
+  "apiVersion": "extensions/v1beta1",
+  "metadata": {
+    "selfLink": "/apis/extensions/v1beta1/ingresses",
+    "resourceVersion": "58845"
+  },
+  "items": [
+    {
+      "metadata": {
+        "name": "the-ingress",
+        "namespace": "default",
+        "selfLink": "/apis/extensions/v1beta1/namespaces/default/ingresses/the-ingress",
+        "uid": "6c1466d7-813e-11e7-b89b-080027a996b8",
+        "resourceVersion": "58840",
+        "generation": 1,
+        "creationTimestamp": "2017-08-14T22:17:55Z"
+      },
+      "spec": {
+        "rules": [
+          {
+            "http": {
+              "paths": [
+                {
+                  "path": "/some-path",
+                  "backend": {
+                    "serviceName": "echo",
+                    "servicePort": 1010
+                  }
+                },
+                {
+                  "path": "/other-path",
+                  "backend": {
+                    "serviceName": "echo",
+                    "servicePort": 2021
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      },
+      "status": {
+        "loadBalancer": {}
+      }
+    }
+  ]
+}"""
+
+  val annotationClass = "linkerd"
+
+  def mkIngressApiServiceReturning(response: String) = Service.mk[Request, Response] {
+    case req if req.uri.contains("/apis/extensions/v1beta1/ingresses") =>
+      val rsp = Response()
+      rsp.content = Buf.Utf8(response)
+      Future.value(rsp)
+    case req =>
+      fail(s"unexpected request for [${req.uri}]: $req")
+  }
+
+  test("builds cache with the existing ingress if only one configured") {
+    val service = mkIngressApiServiceReturning(ingressResourceListWithManyAnnotatedIngresses)
+    val cache = new IngressCache(None, service, annotationClass)
+    assert(await(cache.matchPath(host, "/istio-only")).isEmpty)
+    assert(await(cache.matchPath(host, "/ngix-only")).isEmpty)
```

Comment:
I'm a little confused why this path isn't matched, since I don't see a "kubernetes.io/ingress.class" specified for nginx-ingress, and we should still use the ingress if no ingress.class is specified.